tests/test_json_to_parquet.py

```diff
@@ -0,0 +1,531 @@
+        s3_client = boto3.client("s3")
+        file_basename = os.path.basename(path)
+        s3_key = os.path.relpath(
+                os.path.join(table_location, file_basename),
+                f"s3://{s3_bucket}"
+        )
+        s3_client.upload_file(
+                Filename=path,
+                Bucket=s3_bucket,
+                Key=s3_key
+        )
+
+    @pytest.fixture()
+    def glue_test_data(self, glue_flat_table_location, glue_nested_table_location,
+                       artifact_bucket, datadir):
+        for root, _, files in os.walk(datadir):
+            for basename in files:
+                if "Flat" in basename:
+                    this_table_location=glue_flat_table_location
+                elif "Nested" in basename:
+                    this_table_location=glue_nested_table_location
+                else:
+                    raise ValueError("Found test data that is neither flat nor nested.")
+                absolute_path = os.path.join(root, basename)
+                self.upload_test_data_to_s3(
+                        path=absolute_path,
+                        s3_bucket=artifact_bucket,
+                        table_location=this_table_location
+                )
+
+    @pytest.fixture(scope="class")
+    def glue_crawler_role(self, namespace):
+        iam_client = boto3.client("iam")
+        role_name=f"{namespace}-pytest-crawler-role"
+        glue_service_policy_arn = "arn:aws:iam::aws:policy/service-role/AWSGlueServiceRole"
+        s3_read_policy_arn = "arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess"
+        glue_crawler_role = iam_client.create_role(
+                RoleName=role_name,
+                AssumeRolePolicyDocument=json.dumps({
+                    "Version": "2012-10-17",
+                    "Statement": [
+                        {
+                            "Effect": "Allow",
+                            "Principal": {
+                                "Service": ["glue.amazonaws.com"]
+                            },
+                            "Action": ["sts:AssumeRole"]
+                        }
+                    ]
+                }),
+        )
+        iam_client.attach_role_policy(
+                RoleName=role_name,
+                PolicyArn=glue_service_policy_arn
+        )
+        iam_client.attach_role_policy(
+                RoleName=role_name,
+                PolicyArn=s3_read_policy_arn
+        )
+        yield glue_crawler_role["Role"]["Arn"]
+        iam_client.detach_role_policy(
+                RoleName=role_name,
+                PolicyArn=glue_service_policy_arn
+        )
+        iam_client.detach_role_policy(
+                RoleName=role_name,
+                PolicyArn=s3_read_policy_arn
+        )
+        iam_client.delete_role(RoleName=role_name)
+
+    @pytest.fixture()
+    def glue_crawler(self, glue_database, glue_database_name, glue_database_path, glue_flat_table,
```

Comment:
Could use moto here: https://docs.getmoto.org/en/latest/docs/services/glue.html

Reply:
This is for setting up the test environment on AWS (e.g., uploading test data, creating Glue resources, running crawler to add partitions to Glue tables) so we don't want to use a mock here.